src/mon/MDSMonitor.cc

```diff
@@ -144,7 +144,9 @@ void MDSMonitor::update_from_paxos(bool *need_bootstrap)
   // new map
   dout(4) << "new map" << dendl;
   print_map(fsmap, 0);
-  fsmap.sanity();
+  if (!g_conf->mon_mds_skip_sanity) {
+    fsmap.sanity();
+  }
```

Comment:
We want to do the config check in FSMap itself; Don't Repeat Yourself and that way any future callers don't need to remember to check it.